dbms/src/Server/Client.cpp

```
@@ -197,7 +198,21 @@ class Client : public Poco::Util::Application
 
             default_database = config.getString("database", "");
             user = config.getString("user", "");
-            password = config.getString("password", "");
+
+            if (config.getBool("ask-password", false))
+            {
+                if (config.has("password"))
+                    std::cerr << "Warning: Ignoring --password option because --ask-password specified." << std::endl;
```

It's better to throw an exception.